### DISCUSSION TITLE
Some Gnome Apps pixelated on HiDPI screen

### DISCUSSION THREAD

**Discussion by nr23730** [hyprwm/Hyprland#4379] [Q&A]
I'm running Hyprland 0.34.0 with Fedora 39 on my Laptop. 
Screen is 2880x1800 and scaling is set to 1.5.

However, several Gnome Apps look very pixelated even though they run native on Wayland (I confirmed using `hyprctl clients`).

Has anyone of you experienced something like this before? Am I just missing a setting or env variable?

![image](https://github.com/hyprwm/Hyprland/assets/20756025/553c83cf-f6b8-4061-9a6f-cb68e607debc)

Apps that **don't** behave *as they should*: Nautilus, Calculator, Dejadup, Gnome-Software

Apps that behave *as they should*: Evolution, System Monitor

Environment variables I consider important for this:
```
XDG_SESSION_TYPE=wayland
WAYLAND_DISPLAY=wayland-1
```

**Comment by micahthomas** [comment]
I struggled with this one as well - I think the pieces that fixed it for me finally is setting the display scaling ratio to 2 and setting `GDK_SCALE` environment variable to 2 as well. I think it was blurry before because GDK had a different factor than the hyprland monitor's.

I have a 4k monitor so I would adapt the following for your scaling factor and see if that helps.

```
monitor=HDMI-A-1,3840x2160@120,0x0,2,vrr,1
#                                  ^ this is the scaling factor for the display
# make sure GDK_SCALE matches the monitor's scaling factor
env = GDK_SCALE,2
env = XCURSOR_SIZE,32
```

**Comment by vaxerski** [comment]
Bug in 0.34.0 fixed in -git.

**Reply by micahthomas** [reply]
Thanks for the fix.

I was able to remove the manual `GDK_SCALE` & monitor scaling argument when using `hyperland-git` from AUR.
Nautilus doesn't look pixelated.
